### ISSUE TITLE
ScrollViewer inside of a Grid column has inconsistent scroll bar visibility

### ISSUE THREAD

**Issue by wgraham17** [AvaloniaUI/Avalonia#2701]
ScrollViewer is not showing a scrollbar when it should (based on its content) if it is inside of a Grid that has columns. 

Simple repro:
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaBugRepro.MainWindow"
        Title="AvaloniaBugRepro">
  <Grid RowDefinitions="*,Auto,*" ColumnDefinitions="*,*,*">
    <Canvas Background="Blue" />
    <Canvas Background="Red" Grid.Row="2" />
    <Canvas Background="Green" Grid.Column="1" />
    <Canvas Background="Purple" Grid.Column="2" />
    <Canvas Background="Yellow" Grid.Row="2" Grid.Column="2" />
    <GridSplitter Grid.Row="1" Grid.Column="1" />
    <ScrollViewer Grid.Row="2" Grid.Column="1" HorizontalScrollBarVisibility="Disabled" VerticalScrollBarVisibility="Auto">
      <WrapPanel>
        <Button Margin="2" Width="128" Height="128">Button 1</Button>
        <Button Margin="2" Width="128" Height="128">Button 2</Button>
        <Button Margin="2" Width="128" Height="128">Button 3</Button>
        <Button Margin="2" Width="128" Height="128">Button 4</Button>
        <Button Margin="2" Width="128" Height="128">Button 5</Button>
        <Button Margin="2" Width="128" Height="128">Button 6</Button>
        <Button Margin="2" Width="128" Height="128">Button 7</Button>
        <Button Margin="2" Width="128" Height="128">Button 8</Button>
        <Button Margin="2" Width="128" Height="128">Button 9</Button>
        <Button Margin="2" Width="128" Height="128">Button 10</Button>
      </WrapPanel>
    </ScrollViewer>
  </Grid>
</Window>
```

Note: Change the `ColumnDefinitions` to `Auto,*,Auto` (effectively making cols 0 and 2 be zero-width) and the scrollviewer behaves as expected.

**Comment by grokys** [comment]
Would you be able to try this on `master` using one of our CI builds (https://github.com/AvaloniaUI/Avalonia/wiki/Using-nightly-build-feed)? I think this should have been fixed by https://github.com/AvaloniaUI/Avalonia/pull/2563

**Comment by grokys** [comment]
Assuming this was fixed by #2563 so closing.
